cdap-master/src/test/java/co/cask/cdap/data/tools/UpgradeToolTest.java

```diff
@@ -24,7 +25,10 @@
 public class UpgradeToolTest {
   @Test
   public void testInjector() throws Exception {
-    new UpgradeTool();
+    Injector upgradeToolInjector = new UpgradeTool().createInjector();
+    // should not throw exception
+    // Test the UpgradeDatasetServiceManager injector creation
+    upgradeToolInjector.getInstance(UpgradeDatasetServiceManager.class);
```

Comment:
Its called in the constructor. I also thought it will not but I  tested it that it does by commenting out a binding and making sure it fails.